### PR TITLE
manager: Remove unecessary wait during the PV or PVC creation

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1092

### CODE REVIEW

**Comment by yasker** [comment]
I don't understand why we're modifying the manager logic? It seems natural to wait for PV/PVC ready before we continue.

**Comment by shuo-wu** [comment]
It's natural to check or wait for the prerequisite satisfied before calling PV/PVC creation API. But it's unnecessary to wait for the volume status update after successfully calling the PV/PVC creation in Longhorn APIs.

**Comment by yasker** [comment]
Basically we're changing the API call behavior here. Before the change, when API call completed, PV state should be `bound`. After the change, we don't wait for the PV/PVC status. It would be up to the caller to verify the PV/PVC status.

It makes sense since we don't really know how long it will take to bound the PV/PVC. Though we're still waiting for the PV to show up as ready. We just to do it to make the life a bit easier for the API caller. But how do we know how long we should wait for the PV to show up? That's a bit inconsistent.

**Comment by shuo-wu** [comment]
1. Although waiting for the PVC ready in the API call can make things easier, it's a blocking call and the possible long wait will hurt the user experience. We need to do a trade-off. Hence I think keeping consistent with related Kubernetes API would be the choice.
2. If Longhorn works fine, `volume.KubernetesStatus` will be updated within 1~2s since the PV creation. Hence waiting for PV ready for 5s in our PVC creation API is enough as long as the PV does be already created.

**Comment by yasker** [comment]
My point is, how do we know it will only take 1-2 seconds to wait for PV? We thought we only need to wait for 5 seconds for PV to bound with PVC before.

**Comment by shuo-wu** [comment]
1. I do not know if the PV is not created/unavailable or the volume Kubernetes status hasn't been updated. For the first two, the longer time interval makes no sense. For the latter, 5s is longhorn enough.
2. Considering the case that volume Kubernetes status hasn't been updated, the wait time is decided by how long the KubernetesPVController will take for a sync loop. And most time-consuming operation in controllers is object update. According to my experience, the object update typically takes 1s if Longhorn works fine. 
3. If we do want to remove this PV wait, the user needs to tell the PV name when calling the PVC creation API. It can avoid the meaningless wait if the PV is unavailable during the call, but it's not intelligent enough.

**Comment by yasker** [comment]
> 1. I do not know if the PV is not created/unavailable or the volume Kubernetes status hasn't been updated. For the first two, the longer time interval makes no sense. For the latter, 5s is longhorn enough.

What's the reason of longer wait time doesn't make sense, or 5s is long enough?

> 2. Considering the case that volume Kubernetes status hasn't been updated, the wait time is decided by how long the KubernetesPVController will take for a sync loop. And most time-consuming operation in controllers is object update. According to my experience, the object update typically takes 1s if Longhorn works fine.

The issue we have in the test doesn't always happen neither. I still can't see how it's guaranteed 5s is enough for PV (and somehow doesn't always work for PVC).

> 3. If we do want to remove this PV wait, the user needs to tell the PV name when calling the PVC creation API. It can avoid the meaningless wait if the PV is unavailable during the call, but it's not intelligent enough.

This reason makes sense. We need to wait for PV to know its name, so we can use it for creating PVC.

File: manager/kubernetes.go [hunk 1]
```diff
@@ -128,30 +111,37 @@ func (m *VolumeManager) PVCCreate(name, namespace, pvcName string) (v *longhorn.
...
 	}
-
 	if pvcName == "" {
 		pvcName = v.Name
 	}
 
-	if ks.PVName == "" {
-		return nil, fmt.Errorf("no PVName set in volume %v kubernetes status %v", name, ks)
+	var pvFound bool
+	for i := 0; i < KubeStatusPollConut; i++ {
```
Comment (yasker) [inline manager/kubernetes.go:124]: Typo on `KubeStatusPollCount`

**Pull request merged**